contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -102,45 +124,200 @@ contract StakeManager is IStakeManager, StakeManagerStorage, Initializable {
+    }
+
+    function validatorReward(uint256 validatorId) public view returns (uint256) {
+        uint256 _validatorReward;
+        if (validators[validatorId].deactivationEpoch == 0) {
+            (_validatorReward, ) = _evaluateValidatorAndDelegationReward(validatorId);
+        }
+        return validators[validatorId].reward.add(_validatorReward).sub(INITIALIZED_AMOUNT);
+    }
+
+    function currentValidatorSetSize() public view returns (uint256) {
+        return validatorState.stakerCount;
+    }
+
+    function currentValidatorSetTotalStake() public view returns (uint256) {
+        return validatorState.amount;
+    }
+
+    function getValidatorContract(uint256 validatorId) public view returns (address) {
+        return validators[validatorId].contractAddress;
+    }
+
+    function isValidator(uint256 validatorId) public view returns (bool) {
+        return
+            _isValidator(
+                validators[validatorId].status,
+                validators[validatorId].amount,
+                validators[validatorId].deactivationEpoch,
+                currentEpoch
+            );
+    }
+
+    /**
+        Governance Methods
+     */
+
+    function setDelegationEnabled(bool enabled) public onlyGovernance {
+        delegationEnabled = enabled;
+    }
+
+    // Housekeeping function. @todo remove later
+    function forceUnstake(uint256 validatorId) external onlyGovernance {
+        _unstake(validatorId, currentEpoch);
+    }
+
+    function setCurrentEpoch(uint256 _currentEpoch) external onlyGovernance {
+        currentEpoch = _currentEpoch;
+    }
+
+    function setStakingToken(address _token) public onlyGovernance {
+        require(_token != address(0x0));
+        token = IERC20(_token);
+    }
+
+    /**
+        @dev Change the number of validators required to allow a passed header root
+     */
+    function updateValidatorThreshold(uint256 newThreshold) public onlyGovernance {
+        require(newThreshold > 0);
+        logger.logThresholdChange(newThreshold, validatorThreshold);
+        validatorThreshold = newThreshold;
+    }
+
+    function updateCheckPointBlockInterval(uint256 _blocks) public onlyGovernance {
+        require(_blocks > 0, "incorrect value");
+        checkPointBlockInterval = _blocks;
+    }
+
+    function updateCheckpointReward(uint256 newReward) public onlyGovernance {
+        require(newReward > 0);
+        logger.logRewardUpdate(newReward, CHECKPOINT_REWARD);
+        CHECKPOINT_REWARD = newReward;
+    }
+
+    // New implementation upgrade
+
+    function migrateValidatorsData(
+        uint256 validatorIdFrom,
+        uint256 validatorIdTo
+    ) public onlyOwner {
+        for (uint256 i = validatorIdFrom; i < validatorIdTo; ++i) {
+            ValidatorShare contractAddress = ValidatorShare(validators[i].contractAddress);
+            if (contractAddress != ValidatorShare(0)) {
+                validators[i].delegatorsReward = contractAddress.validatorRewards_deprecated();
```

Comment:
No need add `INITIALIZED_AMOUNT` to `validatorRewards` or `delegatedAmount`?

Reply:
Fixed.